plugin/src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBaseExtension.kt

```diff
@@ -123,12 +123,12 @@ abstract class MavenPublishBaseExtension(
       project.gradleSigning.useInMemoryPgpKeys(inMemoryKeyId, inMemoryKey, inMemoryKeyPassword)
     }
 
-    // TODO: replace with the following line after https://github.com/gradle/gradle/issues/21857 is fixed
-    //  project.gradleSigning.sign(project.gradlePublishing.publications)
-    project.gradlePublishing.publications.withType(MavenPublication::class.java).all { publication ->
-      val task = project.tasks.findByName("sign${publication.name.capitalize()}Publication")
-      if (task == null) {
-        project.gradleSigning.sign(publication)
+    if (GradleVersion.current() < GradleVersion.version("8.0")) {
+      project.mavenPublications { publication ->
+        val task = project.tasks.findByName("sign${publication.name.capitalize()}Publication")
+        if (task == null) {
+          project.gradleSigning.sign(publication)
+        }
```

Comment:
Issue is fixed in 8.0 so we don't need to workaround that causes eager task creation anymore